updateUi.py

```diff
@@ -0,0 +1,92 @@
+from tkinter import *
+from GameData import GameData
+from State import State
+from StatEnum import Stat
+from CharacterInstance import CharacterInstance
+from UserLogs import UserLogs
+
+
+class updateUi(Frame):
+    def __init__(self, user_logs, master=None):
+        Frame.__init__(self, master)
+        self.charattribute = []
+        self.createwidgets()
+        self.user_logs = user_logs
+
+    def update_f(self):
+
+        # 1. select character instance from UserLogs
+        char_instance = self.user_logs.get_char_instance(str(self.charattribute[0].get()))
+
+        # 2. Create new State
+        stat_dict = {Stat.HP: int(self.charattribute[3].get()), Stat.Str: int(self.charattribute[4].get()),
+                     Stat.Mag: int(self.charattribute[5].get()), Stat.Skl: int(self.charattribute[6].get()),
+                     Stat.Spd : int(self.charattribute[7].get()), Stat.Lck : int(self.charattribute[8].get()),
+                     Stat.Def : int(self.charattribute[9].get()), Stat.Res : int(self.charattribute[10].get()), }
+        state = State(int(self.charattribute[2].get()), str(self.charattribute[1].get()), stat_dict)
+
+        # 3. Add state to character instance
+        char_instance.add_new_state(state)
+
+        # 4. Add character Instance to user logs
+        self._user_logs.update_logs(char_instance)
+
+    def createlabels(self, charachter_name, grideN):
```

Comment:
create_labels(self, character_name, gridN)

"""
 create_labels creates a label for a UI element

Args:
character_name : str that specifies the name of the character
gridN : int that specifies the number of rows
"""